lib/internal/bootstrap_node.js

```diff
@@ -697,4 +700,36 @@
   };
 
   startup();
+
+  function makePrimordials() {
+    const ReflectApply = Reflect.apply;
+
+    // This function is borrowed from the function with the same name on V8
+    // Extras' `utils` object. V8 implements Reflect.apply very efficiently in
+    // conjunction with the spread syntax, such that no additional special case
+    // is needed for function calls w/o arguments.
+    // Refs: https://git.io/vAOyO
+    function uncurryThis(func) {
+      return (thisArg, ...args) => ReflectApply(func, thisArg, args);
+    }
+
+    return {
+      uncurryThis,
+
+      ArrayJoin: uncurryThis(Array.prototype.join),
+      ArrayMap: uncurryThis(Array.prototype.map),
+
+      FunctionBind: uncurryThis(Function.prototype.bind),
+
+      ObjectIsPrototypeOf: uncurryThis(Object.prototype.isPrototypeOf),
+      ObjectHasOwnProperty: uncurryThis(Object.prototype.hasOwnProperty),
+      ObjectKeys: Object.keys,
```

Comment:
I guess if we go this way, we might want to add all Object properties?

Reply:
This is more or less why I'm not enthusiastic about this work.